cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -506,36 +321,47 @@ func Run(cfg *completedProxyRunOptions) error {
 	return nil
 }
 
-// Returns intiliazed config, allows local usage (outside cluster) based on provided kubeconfig or in-cluter
-func initKubeConfig(kcLocation string) (*rest.Config, error) {
-	if kcLocation != "" {
-		kubeConfig, err := clientcmd.BuildConfigFromFlags("", kcLocation)
-		if err != nil {
-			return nil, fmt.Errorf("unable to build rest config based on provided path to kubeconfig file: %w", err)
-		}
-		return kubeConfig, nil
+func createKubeRBACProxyConfig(opts *completedProxyRunOptions) (*server.KubeRBACProxyConfig, error) {
+	proxyConfig := server.NewConfig()
+	if err := opts.SecureServing.ApplyTo(&proxyConfig.SecureServing); err != nil {
+		return nil, err
 	}
 
-	kubeConfig, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, fmt.Errorf("cannot find Service Account in pod to build in-cluster rest config: %w", err)
+	if opts.ProxySecureServing != nil {
+		if err := opts.ProxySecureServing.ApplyTo(&proxyConfig.KubeRBACProxyInfo.ProxyEndpointsSecureServing); err != nil {
+			return nil, err
+		}
 	}
 
-	return kubeConfig, nil
-}
+	if err := opts.ProxyOptions.ApplyTo(proxyConfig.KubeRBACProxyInfo); err != nil {
+		return nil, err
+	}
 
-func parseAuthorizationConfigFile(filePath string) (*authz.Config, error) {
-	klog.Infof("Reading config file: %s", filePath)
-	b, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read resource-attribute file: %w", err)
+	if err := opts.LegacyOptions.ApplyTo(proxyConfig.KubeRBACProxyInfo); err != nil {
+		return nil, err
 	}
 
-	configFile := configfile{}
+	return proxyConfig, nil
+}
 
-	if err := yaml.Unmarshal(b, &configFile); err != nil {
-		return nil, fmt.Errorf("failed to parse config file content: %w", err)
-	}
+func prepareSecureServer(
+	ctx context.Context,
+	runGroup *run.Group,
```

Comment:
Handing something like a `run.Group` down is not a good thing to do. it would be better to return the func.